### PR TITLE
[FE] Set Password Page

### PR DESCRIPTION
**Pull request by kimong1**
#112 

* Check [here](https://kimong.learndatabases.dev/setPassword/token)

* Created page to path `/setPassword/:token`
    * Take password input twice, check if they match
    * Set user password
    * Automatically login and redirect to setDBPassword page
       
* Modified `userRoutes.resetPasswordEmail`
    * "Confirmation Email" -> "Set your password"

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -38,6 +38,9 @@ const startServer = async (portNumber) => {
     app.get('/signup', (req, res) => {
       res.render('signup')
     })
+    app.get('/setPassword/:id/:token', (req, res)=>{
```
Comment (songz) [inline src/server.js:41]: `userId` comes from the `token`. so `/setPassword/:token` is enough.

Comment (kimong1) [reply src/server.js:41]: How do you get id from token?

Comment (kimong1) [reply src/server.js]: Fixed

File: views/setPassword.ejs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                token: token,
+                password: password1.value
+            })
+        }).then(r => r.json()).then((r) => {
+            if (r.error) return alert(r.error.message)
+            fetch('/api/session', {
```
Comment (songz) [inline views/setPassword.ejs:69]: `return fetch(....)` You should not have nested promises. Promises goal is to not have these nested promises.

Isn't it better to do `location.href = '/'`?

Comment (kimong1) [reply views/setPassword.ejs:69]: I think it's good to automatically log in the user when they set there password (I have lots of memory thinking that it's annoying that I have to tell the whole user information again just after I told them)
I can do this way
```
submit.addEventListener('click', asyc () => {
  try {
    fetch('/api/passwordReset', {...}).then(...)
  } catch {
  return
  }
  try {
    fetch('/api/session', {...}).then((r) => { location.href='/setDBPassword' })
  } catch {
  ...
  }
})
```

File: views/setPassword.ejs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            <button class="submit">Submit</button>
+        </div>
+    </div>
+</main>
+
+<script>
+    const path = window.location.pathname.split('/')
+    const token = path.pop()
+    console.log('path is', path)
+    console.log('toekn is', token)
```
Comment (songz) [inline views/setPassword.ejs:20]: remove `console.log` please.

Comment (kimong1) [reply views/setPassword.ejs]: Done

**Pull request merged**